sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
I've tried it and it works.

You can't just do `min(b1, b2)` in `merge`. Because `b1` and `b2` can be null. When `b1` is null, just output `b2`, and vice versa. We only call `min` on them when they are both not null.